app/models/identifier_scheme.rb

```diff
@@ -34,4 +37,26 @@ class IdentifierScheme < ActiveRecord::Base
   validates :active, inclusion: { message: INCLUSION_MESSAGE,
                                   in: BOOLEAN_VALUES }
 
+  # ===========================
+  # = Scopes =
+  # ===========================
+
+  scope :active, -> { where(active: true) }
+  scope :for_users, -> { active.where(for_users: true) }
+  scope :for_orgs, -> { active.where(for_orgs: true) }
+  scope :for_plans, -> { active.where(for_plans: true) }
+  scope :authenticatable, -> { active.where(for_auth: true) }
+  scope :by_name, ->(value) { active.where("LOWER(name) = LOWER(?)", value) }
+
+  # ===========================
+  # = Instance Methods =
+  # ===========================
+
+  # The name is used by the OrgSelection Services as a Has key. For example:
+  #    { "ror": "12345" }
+  # so we cannot allow spaces or non alpha characters!
+  def name=(value)
+    super(value&.downcase&.gsub(/[^a-z]/, ""))
+  end
```

Comment:
This is a neat little pattern for overriding the value.  It seems like it will be easier to test than callbacks as well.

Would it be worthwhile adding a validation for this above on the name's value?

Reply:
Hmm. This would happen before any `validate` or `valid?` calls so I think it catches any issues. There is no UI for this either (maybe there should be) so a validation would potentially start causing errors for existing records in some installations. I will add another step to the upgrade rake tasks to convert them to lower case.